app/src/main/java/co/condorlabs/customcomponents/graphics/barsgraph/BarsGraph.kt

```diff
@@ -0,0 +1,266 @@
+        barsBorderCornerRadius, barsBorderCornerRadius,
+        0F, 0F,
+        0F, 0F
+    )
+    private val cornersTwo = floatArrayOf(
+        internalBarsBorderCornerRadius, internalBarsBorderCornerRadius,
+        internalBarsBorderCornerRadius, internalBarsBorderCornerRadius,
+        0F, 0F,
+        0F, 0F
+    )
+
+    // DEFAULT ATTRIBUTES
+    private val defaultHorizontalLinesColor = Color.parseColor("#99c9c9c9")
+    private val defaultLabelTextSize = 42F
+    private val defaultCountTextSize = 92F
+    private val defaultHorizontalLinesStrokeWidth = 2F
+    private val defaultBarsStrokeWidth = 200F
+    private val defaultBarsMargin = 50F
+    private val barsGraphConfigOfExample = BarsGraphConfig(
+        1,
+        listOf(
+            Bar("ONE", 38, Color.GRAY, Color.DKGRAY),
+            Bar("TWO", 10, Color.DKGRAY, Color.BLUE)
+        )
+    )
+
+    init {
+        attrs?.let { setupAttrs(it) }
+    }
+
+    private fun setupAttrs(attrs: AttributeSet) {
+        val typedArray = context.obtainStyledAttributes(
+            attrs,
+            R.styleable.BarsGraph,
+            DEFAULT_STYLE_ATTR,
+            DEFAULT_STYLE_RES
+        )
+        numbersTextSize =
+            typedArray.getDimension(
+                R.styleable.BarsGraph_barsGraphNumbersTextSize,
+                defaultCountTextSize
+            )
+        labelsTextSize =
+            typedArray.getDimension(
+                R.styleable.BarsGraph_barsGraphLabelsTextSize,
+                defaultLabelTextSize
+            )
+        horizontalLinesStrokeWidth = typedArray.getDimension(
+            R.styleable.BarsGraph_barsGraphHorizontalLinesStrokeWidth,
+            defaultHorizontalLinesStrokeWidth
+        )
+        barsStrokeWidth = typedArray.getDimension(
+            R.styleable.BarsGraph_barsGraphBarsStrokeWidth,
+            defaultBarsStrokeWidth
+        )
+        barsMargin =
+            typedArray.getDimension(R.styleable.BarsGraph_barsGraphBarsMargin, defaultBarsMargin)
+        if (typedArray.getBoolean(R.styleable.BarsGraph_barsGraphNumbersBold, false)) {
+            countLabelsPaint.typeface = Typeface.create(Typeface.DEFAULT, Typeface.BOLD)
+        }
+        val labelsColor =
+            typedArray.getColor(R.styleable.BarsGraph_barsGraphLabelsColor, Color.BLACK)
+        val horizontalLinesColor = typedArray.getColor(
+            R.styleable.BarsGraph_barsGraphHorizontalLinesColor,
+            defaultHorizontalLinesColor
+        )
+        typedArray.recycle()
+
+        countLabelsPaint.textSize = numbersTextSize
+        labelsPaint.run { textSize = labelsTextSize; color = labelsColor }
+        horizontalLinesPaint.run {
+            strokeWidth = horizontalLinesStrokeWidth; color = horizontalLinesColor
+        }
+        barsPaint.strokeWidth = barsStrokeWidth
+        setupConfig(barsGraphConfigOfExample)
+    }
+
+    fun setupConfig(barsGraphConfig: BarsGraphConfig) {
```

Comment:
Can be private.

Reply:
never and ever!
This method allows assigning the object with the information to graph.